checks/grpc/check_test.go

```diff
@@ -13,77 +12,63 @@ import (
 	"google.golang.org/grpc/health/grpc_health_v1"
 )
 
-const (
-	addr    = ":8080"
-	service = "HealthTest"
-)
-
-var healthServer *health.Server
-
-func TestMain(m *testing.M) {
-	healthServer = health.NewServer()
-	healthServer.SetServingStatus(service, grpc_health_v1.HealthCheckResponse_SERVING)
-
-	lis, err := net.Listen("tcp", addr)
-	if err != nil {
-		log.Fatalf("error setting up tcp listener: %v", err)
-	}
-
-	server := grpc.NewServer()
-	grpc_health_v1.RegisterHealthServer(server, healthServer)
-
-	go func() {
-		if err := server.Serve(lis); err != nil {
-			log.Fatalf("failed to serve: %v", err)
-		}
-	}()
-
-	defer server.Stop()
-
-	os.Exit(m.Run())
-}
-
-func TestNew_WithServingStatusServing(t *testing.T) {
-	healthServer.SetServingStatus(service, grpc_health_v1.HealthCheckResponse_SERVING)
-
-	check := New(Config{
-		Target:  addr,
-		Service: service,
-		DialOptions: []grpc.DialOption{
-			grpc.WithInsecure(),
+func TestNew(t *testing.T) {
+	for name, tc := range map[string]struct {
+		servingStatus grpc_health_v1.HealthCheckResponse_ServingStatus
+		requireError  bool
+	}{
+		"serving": {
+			servingStatus: grpc_health_v1.HealthCheckResponse_SERVING,
+			requireError:  false,
 		},
-	})
-
-	err := check(context.Background())
-	require.NoError(t, err)
-}
-
-func TestNew_WithServingStatusUnknown(t *testing.T) {
-	healthServer.SetServingStatus(service, grpc_health_v1.HealthCheckResponse_UNKNOWN)
-
-	check := New(Config{
-		Target:  addr,
-		Service: service,
-		DialOptions: []grpc.DialOption{
-			grpc.WithInsecure(),
+		"unknown": {
+			servingStatus: grpc_health_v1.HealthCheckResponse_UNKNOWN,
+			requireError:  true,
 		},
-	})
-
-	err := check(context.Background())
-	require.Error(t, err)
-}
-
-func TestNew_WithServingStatusNotServing(t *testing.T) {
-	healthServer.SetServingStatus(service, grpc_health_v1.HealthCheckResponse_NOT_SERVING)
-
-	check := New(Config{
-		Target:  addr,
-		Service: service,
-		DialOptions: []grpc.DialOption{
-			grpc.WithInsecure(),
+		"not serving": {
+			servingStatus: grpc_health_v1.HealthCheckResponse_NOT_SERVING,
+			requireError:  true,
 		},
-	})
-
-	err := check(context.Background())
-	require.Error(t, err)
+	} {
+		servingStatus := tc.servingStatus
+		requireError := tc.requireError
+
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			const service = "HealthTest"
+
+			healthServer := health.NewServer()
+			healthServer.SetServingStatus(service, servingStatus)
+
+			lis, err := net.Listen("tcp", "localhost:0")
+			require.NoError(t, err)
+
+			server := grpc.NewServer()
+			grpc_health_v1.RegisterHealthServer(server, healthServer)
+
+			go func() {
+				if err := server.Serve(lis); err != nil {
+					log.Fatalf("failed to serve: %v", err)
```

Comment:
using `require.NoError` here could be better than `log.Fatalf`

Reply:
Not really. Neither `log.Fatalf` nor `require.NoError` can't change tests exit code to 1 if called from a goroutine. Actually, we don't care about `err` here because `grpc` server is not our code that we check in these tests. So, I added just `t.Log` to inform.